api/api/admin/__init__.py

```diff
@@ -154,23 +154,40 @@ class IndividualUserPreferencesAdmin(admin.ModelAdmin):
     currently logged-in user's preferences
     """
 
-    can_delete = False
     verbose_name_plural = "My Preferences"
     verbose_name = "My Preferences"
     form = UserPreferencesAdminForm
 
+    def has_add_permission(self, request):
+        """Remove functionality to add new ``UserPreferences`` objects."""
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        """Remove functionality to delete ``UserPreferences`` objects."""
+        return False
+
     def get_queryset(self, request):
+        """
+        Restrict the user's ability to view and change only the
+        ``UserPreferences`` instance mapped to their own ``User`` instance.
```

Comment:
Why not? The function `get_queryset` returns the set of all objects that the current user should be able to see, modify or delete. By the use of `.filter` we have narrowed down the set to the singular instance of `UserPreferences` that is associated with the current user.

Reply:
Why not? The function get_queryset returns the set of all objects that the current user should be able to see, modify or delete. By the use of .filter we have narrowed down the set to the singular instance of UserPreferences that is associated with the current user.

Reply:
Gotcha. It's a comment on specifically filtering just for the user's preferences. To me, with this as the leading comment, it reads as a remark on what `get_queryset` is doing _generally_, rather than the implementation details. "Restrict" being the verb at the start, rather than "retrieve" confused me, as this isn't a permissions function, it's a querying function.

No worries here, just confusing to me, but not a problem.